Core/src/TrackFinding/MeasurementSelector.cpp

```diff
@@ -47,17 +48,18 @@ double MeasurementSelector::calculateChi2(
 
         using ParametersVector = ActsVector<kMeasurementSize>;
 
-        // Take the projector (measurement mapping function)
-        const auto H =
-            projector.template topLeftCorner<kMeasurementSize, eBoundSize>()
-                .eval();
+        FixedSubspaceHelper<eBoundSize, kMeasurementSize> subspaceHelper(
+            std::span<std::uint8_t, kMeasurementSize>(
+                projector.begin(), projector.begin() + kMeasurementSize));
```

Comment:
I understand why this is here, but it looks a bit funky. Could we make this a factory function maybe?